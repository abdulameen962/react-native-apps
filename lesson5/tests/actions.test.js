import * as actions from '../actions';

describe('updateuser returns actions',() => {
    it('returns the correct action',() => {
        expect(actions.updateUser({name:'test name'})).toMatchSnapshot()
    })
    
    it('when passed empty object',() => {
        expect(actions.updateUser({})).toMatchSnapshot()
    })
    
    it('when passed empty name',() => {
        expect(actions.updateUser({name: ''})).toMatchSnapshot()
    })
})

describe('loginUser returns actions',() => {
    const errMessage = 'wrong credentials';
    const fakeToken = 'thisisATestToken';
    const mockLogin = async (username,password) => {
        if (username == 'username' && password == 'password') {
            return fakeToken
        } 
        throw new Error(errMessage)
    }

    it('dispatches the LOG_IN_SENT',async () => {
        const mockDispatch = jest.fn();
        await actions.loginUser('','')(mockDispatch);
        //mockDispatch.mock.calls // all the args  that mock function was invoked on
        // console.log(mockDispatch.mock.calls);
        expect(mockDispatch.mock.calls[0][0]).toEqual({type:actions.LOG_IN_SENT})
        // expect(mockDispatch.mock.calls[1][0]).toEqual({type:actions.LOG_IN_REJECTED})

    })

    it('dispatches the LOG_IN_FULFILLED with correct creds',async () => {
        const mockDispatch = jest.fn();
        // const token = await login('username','password');
        await actions.loginUser('username','password',mockLogin)(mockDispatch);
        // console.log(mockDispatch.mock.calls[1][0]) 
        expect(mockDispatch.mock.calls[1][0]).toEqual({type:actions.LOG_IN_FULFILLED,payload:fakeToken})
        expect(mockDispatch.mock.calls[1]).toMatchSnapshot();
    })

    it('dispatches the LOG_IN_REJECTED with incorrect creds',async () => {
        const mockDispatch = jest.fn();
        await actions.loginUser('','',mockLogin)(mockDispatch);
        expect(mockDispatch.mock.calls[1][0]).toEqual({type:actions.LOG_IN_REJECTED,payload:errMessage})
        expect(mockDispatch.mock.calls[1]).toMatchSnapshot();
    })
})