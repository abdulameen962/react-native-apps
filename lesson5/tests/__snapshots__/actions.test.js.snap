// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loginUser returns actions dispatches the LOG_IN_FULFILLED with correct creds 1`] = `
[
  {
    "payload": "thisisATestToken",
    "type": "LOG_IN_FULFILLED",
  },
]
`;

exports[`loginUser returns actions dispatches the LOG_IN_REJECTED with incorrect creds 1`] = `
[
  {
    "payload": "wrong credentials",
    "type": "LOG_IN_REJECTED",
  },
]
`;

exports[`updateuser returns actions returns the correct action 1`] = `
{
  "payload": {
    "name": "test name",
  },
  "type": "UPDATE_USER",
}
`;

exports[`updateuser returns actions when passed empty name 1`] = `
{
  "payload": {
    "name": "",
  },
  "type": "UPDATE_USER",
}
`;

exports[`updateuser returns actions when passed empty object 1`] = `
{
  "payload": {},
  "type": "UPDATE_USER",
}
`;
